lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -535,16 +535,22 @@ JavaScriptCompiler.prototype = {
     this.resolvePath('data', parts, 0, true, strict);
   },
 
-  resolvePath: function(type, parts, i, falsy, strict) {
+  resolvePath: function(type, parts, startPartIndex, falsy, strict) {
     if (this.options.strict || this.options.assumeObjects) {
       this.push(
-        strictLookup(this.options.strict && strict, this, parts, i, type)
+        strictLookup(
+          this.options.strict && strict,
+          this,
+          parts,
+          startPartIndex,
+          type
+        )
       );
       return;
     }
 
     let len = parts.length;
-    for (; i < len; i++) {
+    for (let i = startPartIndex; i < len; i++) {
```

Comment:
Why declare `i` here instead of using `parts[startPartIndex]` at line 556?

Reply:
I think it is better style not to use the parameter as variable, but instead us an ordinary for-loop with a pattern that everybody understands